patches/content-browser-frame_host-navigation_request.cc.patch

```diff
@@ -1,30 +1,34 @@
 diff --git a/content/browser/frame_host/navigation_request.cc b/content/browser/frame_host/navigation_request.cc
-index 37f6135b22a4472733afa38dbc3bd27ccfb60036..ea00172e369b134f4cd1cd19bf55c2e160d1e12a 100644
+index 37f6135b22a4472733afa38dbc3bd27ccfb60036..b8df278a78d568c9c41a3114c05bb208ef5a66f1 100644
 --- a/content/browser/frame_host/navigation_request.cc
 +++ b/content/browser/frame_host/navigation_request.cc
-@@ -1651,6 +1651,29 @@ void NavigationRequest::CommitNavigation() {
+@@ -1651,6 +1651,33 @@ void NavigationRequest::CommitNavigation() {
      }
      associated_site_instance_id_.reset();
    }
 +
-+  auto* pending_entry =
-+    frame_tree_node_->navigator()->GetController()->GetPendingEntry();
-+  if (pending_entry) {
-+    if (!pending_entry->GetURL().is_empty()) {
-+      common_params_.url = pending_entry->GetURL();
-+    }
-+    if (!pending_entry->GetReferrer().url.is_empty()) {
-+      common_params_.referrer = pending_entry->GetReferrer();
-+    }
-+  } else {
-+    auto* last_committed_entry =
-+      frame_tree_node_->navigator()->GetController()->GetLastCommittedEntry();
-+    if (last_committed_entry) {
-+      if (!last_committed_entry->GetURL().is_empty()) {
-+        common_params_.url = last_committed_entry->GetURL();
++  // With chrome-extension by using Pocket on a page with a youtube iframe,
++  // it can lead to insecure referrer mismatch error.
++  if (!common_params_.url.SchemeIs("chrome-extension")) {
```

Comment:
maybe we should be checking secure/insecure schemes for all redirects? Might be other similar issues somewhere? Isn't there a referrer method that checks that?